examples/c/5_series/gnss_ins/5_series_gnss_ins_example.c

```diff
@@ -1,354 +1,1106 @@
+    while (filter_status.filter_state < MIP_FILTER_MODE_GX5_RUN_SOLUTION_VALID)
+    {
+        // Update the device state
+        // Note: This will update the device callbacks to trigger the filter state change
+        mip_interface_update(
+            &device,
+            0,    // Time to wait
+            false // From command
+        );
+
+        // Fix type state change
+        if (current_fix_type != gnss_fix_info.fix_type)
+        {
+            display_gnss_fix_state(&gnss_fix_info);
+            current_fix_type = gnss_fix_info.fix_type;
+        }
 
-    const mip_descriptor_rate sensor_descriptors[4] = {
-        { MIP_DATA_DESC_SENSOR_TIME_STAMP_GPS, sensor_decimation },
-        { MIP_DATA_DESC_SENSOR_ACCEL_SCALED,   sensor_decimation },
-        { MIP_DATA_DESC_SENSOR_GYRO_SCALED,    sensor_decimation },
-        { MIP_DATA_DESC_SENSOR_MAG_SCALED,     sensor_decimation },
-    };
+        // Filter state change
+        if (current_state != filter_status.filter_state)
+        {
+            display_filter_state(filter_status.filter_state);
+            current_state = filter_status.filter_state;
+        }
+    }
 
-    if(mip_3dm_write_imu_message_format(&device, 4, sensor_descriptors) != MIP_ACK_OK)
-        exit_gracefully("ERROR: Could not set sensor message format!");
+    // Get the start time of the device update loop to handle exiting the application
+    const mip_timestamp loop_start_time = get_current_timestamp();
 
+    mip_timestamp previous_print_timestamp = 0;
 
-    //
-    //Setup GNSS data format to 4 Hz (decimation of 1)
-    //
+    // Device loop
+    // Exit after predetermined time in seconds
+    while (get_current_timestamp() - loop_start_time <= RUN_TIME_SECONDS * 1000)
+    {
+        // Update the device state
+        // Note: This will update the device callbacks to trigger the filter state change
+        mip_interface_update(
+            &device,
+            0,    // Time to wait
+            false // From command
+        );
+
+        // Fix type state change
+        if (current_fix_type != gnss_fix_info.fix_type)
+        {
+            display_gnss_fix_state(&gnss_fix_info);
+            current_fix_type = gnss_fix_info.fix_type;
+        }
 
-    const mip_descriptor_rate gnss_descriptors[1] = {
-        { MIP_DATA_DESC_GNSS_FIX_INFO, 1 }
-     };
+        // Filter state change
+        if (current_state != filter_status.filter_state)
+        {
+            display_filter_state(filter_status.filter_state);
+            current_state = filter_status.filter_state;
+        }
 
-    //GNSS
-    if(mip_3dm_write_gnss_message_format(&device, 1, gnss_descriptors) != MIP_ACK_OK)
-        exit_gracefully("ERROR: Could not set GNSS1 message format!");
+        const mip_timestamp current_timestamp = get_current_timestamp();
 
+        // Print out data based on the sample rate (1000 ms / SAMPLE_RATE_HZ)
+        if (current_timestamp - previous_print_timestamp >= 1000 / SAMPLE_RATE_HZ)
```

Comment:
Not sure what this is supposed to do. The device will stream at the requested rate, so you shouldn't need to check yourself. No updates will happen without a packet being transmitted, so it shouldn't spam the console without this check.